src/error.rs

```diff
@@ -50,3 +50,20 @@ impl Display for HidError {
 }
 
 impl Error for HidError {}
+
+impl From<std::io::Error> for HidError {
+    fn from(e: std::io::Error) -> Self {
+        Self::HidApiError {
```

Comment:
I didn't want to add to the public API here and break error handling that users of the library might already have, but for this it's probably OK since making use of more rust probably means this is going to be a fairly normal error that we'll have to deal with.

Reply:
I've added a variant for this as it's generally useful and we'll need it more as we write more Rust.